src/DrawManager.sol

```diff
@@ -270,121 +309,138 @@ contract DrawManager {
+  function getLastStartDrawAuction() public view returns (StartDrawAuction memory result) {
+    uint length = _startDrawAuctions.length;
+    if (length > 0) {
+      result = _startDrawAuctions[length-1];
+    }
   }
 
-  /// @notice The last auction results.
-  /// @return StartDrawAuctions struct from the last auction.
-  function getLastStartDrawAuction() external view returns (StartDrawAuction memory) {
-    return _lastStartDrawAuction;
+  function getStartDrawAuctionCount() external view returns (uint) {
+    return _startDrawAuctions.length;
   }
 
-  /// @notice Computes the start draw and finish draw rewards.
-  /// @param _startDrawElapsedTime The elapsed time between draw close and startDraw()
-  /// @param _finishDrawElapsedTime The elapsed time between startDraw() and finishDraw()
-  /// @param _totalReserve The total reserve available to allocate rewards from
-  /// @return rewards The computed rewards for the start and finish draw auctions
-  /// @return remainingReserve The remaining reserve after the rewards have been allocated
-  function computeRewards(uint48 _startDrawElapsedTime, uint48 _finishDrawElapsedTime, uint256 _totalReserve) public view returns (uint256[] memory rewards, uint256 remainingReserve) {
-    UD2x18[] memory rewardFractions = new UD2x18[](2);
-    uint rewardPool = _totalReserve > maxRewards ? maxRewards : _totalReserve;
-    rewardFractions[0] = computeStartDrawRewardFraction(_startDrawElapsedTime);
-    rewardFractions[1] = computeFinishDrawRewardFraction(_finishDrawElapsedTime);
-
-    uint totalRewards;
-    (rewards, totalRewards) = RewardLib.rewards(
-      rewardFractions,
-      rewardPool
-    );
-    remainingReserve = _totalReserve - totalRewards;
+  function getStartDrawAuction(uint256 _index) external view returns (StartDrawAuction memory) {
+    return _startDrawAuctions[_index];
+  }
+
+  function _computeFinishDrawReward(
+    uint256 _auctionOpenedAt,
+    uint256 _auctionAt,
+    uint256 _availableRewards
+  ) internal view returns (uint256 reward, UD2x18 fraction) {
+    fraction = computeFinishDrawRewardFraction(_computeElapsedTime(_auctionAt, _auctionOpenedAt));
+    reward = RewardLib.reward(fraction, _availableRewards);
+  }
+
+  function _computeStartDrawRewards(
+    uint256 _firstAuctionOpenedAt,
+    uint256 _lastAuctionAt,
+    uint256 _availableRewards
+  ) internal view returns (uint256[] memory rewards, UD2x18[] memory fractions) {
+    uint length = _startDrawAuctions.length;
+    rewards = new uint256[](length+1);
+    fractions = new UD2x18[](length+1);
+    uint256 previousStartTime = _firstAuctionOpenedAt;
+    for (uint i = 0; i < rewards.length; i++) {
+      uint closedAt = i < length ? _startDrawAuctions[i].closedAt : _lastAuctionAt;
+      fractions[i] = computeStartDrawRewardFraction(_computeElapsedTime(closedAt, previousStartTime));
+      rewards[i] = RewardLib.reward(fractions[i], _availableRewards);
+      previousStartTime = closedAt;
+    }
```

Comment:
It's slightly confusing that the same function is used to compute the actual rewards (for the start draw auctions that have completed) and the projected rewards (for the next start draw auction). These two use cases are not usually called together and can be separated into two different functions for clarity. For example, there are a couple times where this is called with `_lastAuctionAt` set to zero since we are ignoring the projected reward.